server/sockets/cohort.js

```diff
@@ -237,218 +262,139 @@ const updateCohortHeaderStatus = (socket, cohortClientLocks) => {
-
-          listIdsUserWasRemovedFrom.forEach(listId => {
-            io.sockets
-              .to(listChannel(listId))
-              .emit(ListActionTypes.REMOVE_MEMBER_SUCCESS, {
-                listId,
-                userId
-              });
-
-            if (listClients.has(userId)) {
-              const { socketId, viewId } = listClients.get(userId);
-              const isCohortMember = checkIfArrayContainsUserId(
-                cohortMemberIds,
-                userId
-              );
-
-              if (viewId === listId) {
-                io.sockets
-                  .to(socketId)
-                  .emit(ListActionTypes.REMOVED_BY_SOMEONE, {
-                    cohortId,
-                    isCohortMember,
-                    listId
-                  });
-              }
-            }
-          });
-        }
-
-        if (listIdsUserRemained.length > 0) {
-          listIdsUserRemained.forEach(listId => {
-            if (listClients.has(userId)) {
-              const { socketId, viewId } = listClients.get(userId);
-
-              if (viewId === listId) {
-                io.sockets
-                  .to(socketId)
-                  .emit(ListActionTypes.MEMBER_UPDATE_SUCCESS, {
-                    isCurrentUserUpdated: true,
-                    isGuest: true,
-                    listId,
-                    userId
-                  });
-              }
-            }
-
-            io.sockets
-              .to(listChannel(listId))
-              .emit(ListActionTypes.MEMBER_UPDATE_SUCCESS, {
-                isCurrentUserUpdated: false,
-                isGuest: true,
-                listId,
-                userId
-              });
-          });
-        }
-      }
     });
+  } catch {
+    // Ignore error
+  }
+
+  return Promise.resolve();
 };
 
-const deleteCohort = (io, allCohortsClients) => data => {
-  const { cohortId, owners } = data;
+const deleteCohort = io => data => {
+  const { cohortId, memberIds, performer } = data;
 
   io.sockets
     .to(cohortChannel(cohortId))
-    .emit(CohortActionTypes.REMOVE_WHEN_COHORT_UNAVAILABLE, cohortId);
+    .emit(CohortActionTypes.DELETE_SUCCESS, { cohortId, performer });
+
+  io.sockets
+    .to(cohortMetaDataChannel(cohortId))
+    .emit(CohortActionTypes.DELETE_SUCCESS, { cohortId });
 
-  owners.forEach(id => {
-    const ownerId = id.toString();
+  memberIds.forEach(async id => {
+    const memberId = id.toString();
 
-    if (allCohortsClients.has(ownerId)) {
-      const { socketId } = allCohortsClients.get(ownerId);
+    try {
+      const socketIds = await getUserSockets(memberId);
 
-      io.sockets
-        .to(socketId)
-        .emit(CohortActionTypes.DELETE_SUCCESS, { cohortId });
+      socketIds.forEach(socketId =>
+        io.sockets
+          .to(socketId)
+          .emit(CohortActionTypes.LEAVE_ROOM, cohortMetaDataChannel(cohortId))
+      );
+    } catch {
+      // Ignore errors
     }
   });
+
+  return Promise.resolve();
```

Comment:
here and everywhere we use this Promise.resolve()

Reply:
In sockets' method where we get data from DB - yes, eg. we often get socketIds from DB